components/camel-kudu/src/main/java/org/apache/camel/component/kudu/KuduProducer.java

```diff
@@ -0,0 +1,136 @@
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.kudu;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.apache.camel.Exchange;
+import org.apache.camel.impl.DefaultProducer;
+import org.apache.camel.util.ObjectHelper;
+import org.apache.kudu.ColumnSchema;
+import org.apache.kudu.Schema;
+import org.apache.kudu.client.CreateTableOptions;
+import org.apache.kudu.client.Insert;
+import org.apache.kudu.client.KuduClient;
+import org.apache.kudu.client.KuduException;
+import org.apache.kudu.client.KuduScanner;
+import org.apache.kudu.client.KuduTable;
+import org.apache.kudu.client.PartialRow;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The Kudu producer.
+ *
+ * @see org.apache.camel.component.kudu.KuduEndpoint
+ */
+public class KuduProducer extends DefaultProducer {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KuduProducer.class);
+
+    private final KuduEndpoint endpoint;
+
+    public KuduProducer(KuduEndpoint endpoint) {
+        super(endpoint);
+
+        if (ObjectHelper.isEmpty(endpoint.getKuduClient())) {
+            throw new IllegalArgumentException("Can't create a producer when the database connection is null");
+        }
+        this.endpoint = endpoint;
+    }
+
+    @Override
+    public void process(Exchange exchange) throws Exception {
+        String table = endpoint.getTableName();
+        switch (endpoint.getOperation()) {
+        case INSERT:
+            doInsert(exchange, table);
+            break;
+        case CREATE_TABLE:
+            doCreateTable(exchange, table);
+            break;
+        case SCAN:
+            doScan(exchange, table);
+            break;
+        default:
+            throw new IllegalArgumentException("The operation " + endpoint.getOperation() + " is not supported");
+        }
+    }
+
+    private void doInsert(Exchange exchange, String tableName) throws KuduException {
+        KuduClient connection = endpoint.getKuduClient();
+        KuduTable table = connection.openTable(tableName);
+
+        Insert insert = table.newInsert();
+        PartialRow row = insert.getRow();
+
+        Map<?, ?> rows = exchange.getIn().getBody(Map.class);
+
+        for (Map.Entry<?, ?> entry : rows.entrySet()) {
+            String key = (String) entry.getKey();
+            Object value = entry.getValue();
+            switch (value.getClass().toString()) {
+            case "class java.lang.String":
```

Comment:
Isn't a better way to express this, it's really ugly :-(

Reply:
I thought the same when I saw it :) Will look for something better. It looks like a list of if-elses with instanceof is an antipattern.